Python 3.8 doesn't build on 11.0

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

<!-- To help us debug your issue, please complete these sections: -->

## What you were trying to do (and why)

`python@3.8` was in the dependency tree of software I was installing, but failed to build from source on Big Sur.

## What happened (include command output)

Installation fails; logs here: https://gist.github.com/mistydemeo/658ea7401c00b2b8dccee9f067ac2c0e

Specifically, the error is here:

```
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.16.sdk -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.16.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.16.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers   -std=c99 -Wextra -Wno-unused-result -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration  -I./Include/internal  -I. -I./Include    -DPy_BUILD_CORE_BUILTIN  -DPy_BUILD_CORE_BUILTIN -I./Include/internal -c ./Modules/posixmodule.c -o Modules/posixmodule.o
./Modules/posixmodule.c:9141:20: error: variable has incomplete type 'struct sf_hdtr'
    struct sf_hdtr sf;
                   ^
./Modules/posixmodule.c:9141:12: note: forward declaration of 'struct sf_hdtr'
    struct sf_hdtr sf;
           ^
1 error generated.
```

This happened on macOS 11.0 beta 1 for Intel using the CLT.

## What you expected to happen

Build succeeds.

## Step-by-step reproduction instructions (by running `brew install` commands)

1. `brew install python@3.8`

--- a/Formula/python@3.8.rb
+++ b/Formula/python@3.8.rb
@@ -4,6 +4,13 @@ class PythonAT38 < Formula
   url "https://www.python.org/ftp/python/3.8.3/Python-3.8.3.tar.xz"
   sha256 "dfab5ec723c218082fe3d5d7ae17ecbdebffa9a1aea4d64aa3a2ecdd2e795864"
 
+  # Remove for > 3.8.3
+  # Upstream commit from 2020-06-25 "Support macOS 11 when building"
+  patch do
+    url "https://github.com/python/cpython/commit/8ea6353.patch?full_index=1"
+    sha256 "c47680c85f201f5830bf71741f09ece031b99386040f3c70b20190b4c47fb81d"
+  end
+
   bottle do
     sha256 "67cceb372f3b0f45ff90beb70d6e92ce1416fdd0b3df4df95643fa21451b4fec" => :catalina
     sha256 "6173014910a8902f0601a1b9e1c55bea85d25193bd64180f363e9f37c2d23244" => :mojave

